About/FAQ.md

```diff
@@ -133,6 +133,12 @@ Should I validate the TPMTOTP on every boot
 
 Probably. I want to make it also do it at S3.  [See Heads issue #69](https://github.com/osresearch/heads/issues/69)
 
+How is the TPM used in heads?
+----
+
+Stores secret something?  Could also have a disk unlock key.
```

Comment:
Originally, Heads base was coreboot 4.8.1 with a [measured boot patch applied on top of it](https://github.com/osresearch/heads/blob/master/patches/coreboot-4.8.1/0000-measuredboot.patch). 

In newer versions, coreboot integrated measured boot, and that patch was modified to respect the same [measuring scheme then coreboot](https://doc.coreboot.org/security/vboot/index.html). The PCRs used in Heads extends the ones of coreboot by also measuring the state of the Recovery shell (if going into recovery shell, the measurements are extended and invalidated). The user related files are also measured (GPG public key, config.user overlay). [You should refer here to produce this doc and extend the one there or refer to the one there and extend there](https://osresearch.net/Keys/#tpm-pcrs). Duplicating doc duplicates maintenance of such doc.

Heads takes those measurements to calculate a secret, which upon sealing (TOTP/HOTP) is "challenged" against an external device. 

In the case of TOTP, this output takes the form of a Qr code, to be scanned on a smartphone to remote attest the firmware state at each boot on that smartphone. This requires the user to verify manually on his phone.

In the case of HOTP against Heads' supported USB Security dongles (Nitrokey Pro, Nitrokey Storage, Librem Key), in presence of a TPM, those measurements are used to challenge the USB Security dongle with that value.

In the case of the user having setuped a Disk Unlock Key (a secondary LUKS header slot) through setting a default boot option, the a TPM defined NV memory region can release a randomized Disk Unlock Key if: the TPM measurements are valid and the passphrase to unlock the TPM NV space is valid. In that case, the TPM will release the Disk Unlock Key, embedded into a initrd additional file which is passed to the OS. The Disk Unlock Key passphrase can be changed by setting a new boot default, and requires the user to type his Disk Recovery Key passphrase and then renew/change his Disk Unlock Key passphrase. This changes /boot content, and will also ask for the GPG User PIN to sign the changes.